tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -426,21 +427,27 @@ def find_kernel_bounds(string):
     return output_string
 
 
-def find_parenthesis_end(input_string, start):
+def find_closure_group(input_string, start, group):
+    """Generalization for finding a balancing closure group
+
+    e.g. if group = ["(", ")"], then finds the first balanced parantheses.
+         if group = ["{", "}"], then finds the first balanced bracket.
```

Comment:
OK, but what does it return in this case?

Here, an example is worth a million words:

```
>>> find_closure_group("foo(bar)", 0, ["(", ")"])
????
```

Reply:
@Jorghi12 can you comment?